chiselFrontend/src/main/scala/chisel3/internal/Namer.scala

```diff
@@ -51,22 +77,14 @@ class NamingContext {
     * prefixed with the name given to the reference object, if the reference object is named in the
     * scope of this context.
     */
-  def add_descendant(ref: AnyRef, descendant: NamingContext) {
-    if (!descendants.containsKey(ref)) {
-      descendants.put(ref, ListBuffer[NamingContext]())
+  def addDescendant(ref: Any, descendant: NamingContext) {
+    ref match {
+      case ref: AnyRef =>
+        descendants.getOrElseUpdate(ref, ListBuffer[NamingContext]()) += descendant
+      case _ => anonymousDescendants += descendant
     }
-    descendants.get(ref) += descendant
-  }
-
-  def add_anonymous_descendant(descendant: NamingContext) {
-    anonymousDescendants += descendant
   }
 
-  /** Suggest a name (that will be propagated to FIRRTL) for an object, then returns the object
-    * itself (so this can be inserted transparently anywhere).
-    * Is a no-op (so safe) when applied on objects that aren't named, including non-Chisel data
-    * types.
-    */
   def name[T](obj: T, name: String): T = {
```

Comment:
Is there a reason scaladoc is being deleted?

Reply:
Moved to the interface definition in the trait